src/main/kotlin/lotto/agency/LottoJudge.kt

```diff
@@ -8,7 +8,7 @@ class LottoJudge {
             .toMutableMap()
 
         val determinedLottoWinnings = lottoTickets
-            .map { LottoWinning.of(it.countMatchWonLottoTicket(wonLottoTicket)) }
+            .map { LottoWinning.of(it.countMatchWonLottoTicket(wonLottoTicket), it.isMatchedBonus(wonLottoTicket.bonus!!)) }
```

Comment:
확장함수를 체이닝으로 걸게 되면 매 함수마다 새로운 컬렉션이 생성이 되는데요,
.asSequence를 활용하면 위 문제가 해결됩니다.

```suggestion
            .asSequence()
            .map { LottoWinning.of(it.countMatchWonLottoTicket(wonLottoTicket), it.isMatchedBonus(wonLottoTicket.bonus!!)) }
            ....
```
마지막에는 toList() 등 컬렉션으로 변경하는 체이닝 함수로 마무리 하면 됩니다. 🙂